app/controllers/adjustments_controller.rb

```diff
@@ -13,44 +11,45 @@ def index
 
   # GET /adjustments/1
   # GET /adjustments/1.json
-  def show; end
+  def show
+    @adjustment = current_organization.adjustments.find(params[:id])
+  end
 
   # GET /adjustments/new
   def new
     @adjustment = current_organization.adjustments.new
     @adjustment.line_items.build
-    @storage_locations = current_organization.storage_locations
-    @items = current_organization.items.alphabetized
+    load_form_collections
   end
 
   # POST /adjustments
   def create
     @adjustment = current_organization.adjustments.new(adjustment_params)
-    @storage_locations = current_organization.storage_locations
-    @items = current_organization.items.alphabetized
-
-    if @adjustment.valid?
-      @adjustment.storage_location.adjust!(@adjustment)
 
-      if @adjustment.save
-        redirect_to adjustment_path(@adjustment), notice: "Adjustment was successfully created."
-      else
-        flash[:error] = @adjustment.errors.collect { |model, message| "#{model}: " + message }.join("<br />".html_safe)
-        render :new
+    if @adjustment.valid? && @adjustment.save
+      increasing_adjustment, decreasing_adjustment = @adjustment.split_difference
+      ActiveRecord::Base.transaction do
```

Comment:
I love this! I hadn't used ActiveRecord::Base.transaction and I looked it up. Super nice!
https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html